csrc/quantization/compressed_tensors/int8_quant_kernels.cu

```diff
@@ -103,134 +105,162 @@ static inline __device__ int8_t int32_to_int8(int32_t x) {
 
 namespace vllm {
 
-template <typename scalar_t, typename scale_type>
+template <typename scalar_t, typename scale_t>
 __global__ void static_scaled_int8_quant_kernel(
-    scalar_t const* __restrict__ input, int8_t* __restrict__ out,
-    scale_type const* scale_ptr, const int hidden_size) {
-  int const tid = threadIdx.x;
-  int64_t const token_idx = blockIdx.x;
-  scale_type const scale = *scale_ptr;
-
-  // Must be performed using 64-bit math to avoid integer overflow.
-  out += token_idx * hidden_size;
-  input += token_idx * hidden_size;
-
-  for (int i = tid; i < hidden_size; i += blockDim.x) {
-    out[i] = float_to_int8_rn(static_cast<float>(input[i]) / scale);
-  }
+    const scalar_t* __restrict__ input, int8_t* __restrict__ output,
+    const scale_t* scale_ptr, const int hidden_size) {
+  int tid = threadIdx.x;
+  int stride = blockDim.x;
+  float scale = *scale_ptr;
+
+  const scalar_t* row_in = input + blockIdx.x * hidden_size;
+  int8_t* row_out = output + blockIdx.x * hidden_size;
```

Comment:
I think we should keep the 64-bit math for safety, no?

Reply:
https://github.com/vllm-project/vllm/pull/9391
Found the original issue, fixed, thanks for pointing out!